tools/ci/benchmark-travis-pr.sh

```diff
@@ -23,20 +26,35 @@ run_asv () {
     pip install -e .
     git show --no-patch --format="%H (%s)"
     configure_asv
-    asv run -E existing --set-commit-hash $(git rev-parse HEAD)
+    asv run -E existing --set-commit-hash "$(git rev-parse HEAD)"
+}
+
+#
+# High level actions
+#
+
+setup () {
+    pip install asv
+    asv machine --yes
 }
 
-pip install asv
-asv machine --yes
+run () {
+    git update-ref refs/bm/pr HEAD
+    # We know this is a PR run. The branch is a GitHub refs/pull/*/merge ref, so
+    # the current target that this PR will be merged into is HEAD^1.
+    git update-ref refs/bm/merge-target HEAD^1
 
-git update-ref refs/bm/pr HEAD
-# We know this is a PR run. The branch is a GitHub refs/pull/*/merge ref, so
-# the current target that this PR will be merged into is HEAD^1.
-git update-ref refs/bm/merge-target HEAD^1
+    run_asv
 
-run_asv
+    git checkout --force refs/bm/merge-target
+    run_asv
+}
 
-git checkout --force refs/bm/merge-target
-run_asv
+compare () {
+    asv compare refs/bm/merge-target refs/bm/pr
+}
 
-asv compare refs/bm/merge-target refs/bm/pr
+case "$action" in
+    '') setup && run && compare;;
+    *) eval "$action";;
```

Comment:
Please don't `eval` arbitrary input.  There are only a few valid cases, and I don't see any problem with having an arm for each one.

Reply:
Yeah... Better be safe than sorry, will redo with a dedicated arm for all three and the eval ;-)